libraries/resource_providers/api200/user_provider.rb

```diff
@@ -0,0 +1,50 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require_relative '../../resource_provider'
+
+module OneviewCookbook
+  module API200
+    # User API200 provider
+    class UserProvider < ResourceProvider
+      def initialize(context)
+        super
+        n = @item.data.delete('name')
+        @item['userName'] ||= n
+      end
+
+      def create_or_update
+        pw = @item.exists? ? @item.data.delete('password') : nil
+        super
+        if pw
+          opts = {
+            api_version: @item.client.api_version,
+            url: @item.client.url,
+            user: @item['userName'],
+            password: pw,
+            ssl_enabled: @item.client.ssl_enabled,
+            log_level: :error
+          }
+          begin
+            OneviewCookbook::Helper.build_client(opts)
```

Comment:
This is the only way I could come up with to check if the password was valid/up-to-date